pkg/selector/selector.go

```diff
@@ -171,6 +171,7 @@ func SelectPods(ctx context.Context, c client.Client, r client.Reader, selector
 			}
 
 			pods = append(pods, podList.Items...)
+			podList.Items = nil
```

Comment:
Metadata `Label` seem to not be updated whenever we fetch pods from multiple namespace. Will submit a bug/dig deeper into the reasons in PR that will follow